providers/pool.rb

```diff
@@ -232,6 +236,38 @@ def configure
       shell_out!(cmd)
       @new_resource.updated_by_last_action(true)
     end
+    if is_new_start_mode
+      was_updated = true
+      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
+      cmd << " /startMode:#{@new_resource.start_mode.to_s}"
+      Chef::Log.debug(cmd)
+      shell_out!(cmd)
+      @new_resource.updated_by_last_action(true)
+    end
+    if is_new_auto_start
+      was_updated = true
+      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
+      cmd << " /autoStart:#{@new_resource.auto_start.to_s}"
```

Comment:
Can you clarify are you wanting to combine the commands down to one (totally possible) or simply factorize the code (IE make a function that does what i'm doing in each of these if's, which is likely going to be needed soon since i'm duplicating a lot of code at this point and the more properties we add the more that this will be important